tests/unit/igraph_percolation.c

```diff
@@ -0,0 +1,198 @@
+/*
+    IGraph library.
+    Copyright (C) 2025  The igraph development team <igraph@igraph.org>
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+#include "test_utilities.h"
+
+igraph_error_t percolate_b(igraph_t *graph, igraph_vector_int_t *edge_indices) {
+    igraph_vector_int_t outputs;
+    IGRAPH_VECTOR_INT_INIT_FINALLY(&outputs, 0);
+
+    IGRAPH_CHECK(igraph_bond_percolation(graph, &outputs, edge_indices));
+
+    print_vector_int(&outputs);
+
+    igraph_vector_int_destroy(&outputs);
+    IGRAPH_FINALLY_CLEAN(1);
+    return IGRAPH_SUCCESS;
+}
```

Comment:
In tests, we don't need to implement error handling.

You don't have to remove this here, but it's not really necessary.